view/frontend/layout/checkout_index_index.xml

```diff
@@ -16,19 +16,22 @@
                 <argument name="data_layer_processors" xsi:type="array">
                     <item name="checkout" xsi:type="object">Yireo\GoogleTagManager2\DataLayer\Processor\Checkout</item>
                 </argument>
-
-                <argument name="data_layer_events" xsi:type="array">
-                    <item name="begin_checkout" xsi:type="array">
-                        <item name="event" xsi:type="string">begin_checkout</item>
-                        <item name="ecommerce" xsi:type="array">
-                            <item name="items" xsi:type="object">Yireo\GoogleTagManager2\DataLayer\Tag\Cart\CartItems
-                            </item>
-                        </item>
-                    </item>
-                </argument>
             </arguments>
         </referenceBlock>
 
+        <referenceContainer name="yireo_googletagmanager2.data-layer.container">
+            <block
+                    name="yireo_googletagmanager2.script-begin-checkout"
+                    template="Yireo_GoogleTagManager2::luma/script-begin-checkout.phtml"
+                    ifconfig="googletagmanager2/settings/enabled"
+                    after="yireo_googletagmanager2.data-layer"
+            >
+                <arguments>
+                    <argument name="begin_checkout_event" xsi:type="object">Sanders\GoogleTagManager2\DataLayer\Event\BeginCheckout</argument>
```

Comment:
@koentjeh Minor detail, I can fix it myself (but busy for the current week): Could you change the namespace here into `Yireo`?